### PR TITLE
Make Scorm XBlock compatible with edX completion tool

### PR DESCRIPTION
**Pull request by tehreem-sadat**

### CODE REVIEW

File: scormxblock/scormxblock.py [hunk 1]
```diff
@@ -33,6 +35,8 @@
 
 
 class ScormXBlock(XBlock):
+    has_custom_completion = True
```
Comment (ziafazal) [inline scormxblock/scormxblock.py]: define main class like this

```
ScormXBlock(XBlock, CompletableXBlockMixin):
```
Using [CompletableXBlockMixin](https://github.com/edx/XBlock/blob/master/xblock/completable.py) gives you `emit_completion` method so you don't need to validate completion and we don't need to set these
```
has_custom_completion = True
completion_mode = XBlockCompletionMode.COMPLETABLE
```

File: scormxblock/scormxblock.py [hunk 2]
```diff
@@ -289,6 +293,7 @@ def publish_grade(self):
                     'value': self.lesson_score,
                     'max_value': self.weight,
                 })
+        self.publish_completion()
```
Comment (ziafazal) [inline scormxblock/scormxblock.py]: Completion status of xblock should not depend on whether it is scorable or not. I think at the end of `scorm_set_value` method where we are pulling completion status we should add these lines to mark completion
```
completion_status = self.get_completion_status()
if completion_status in ["passed", "failed", "completed"]:
    self. emit_completion(1.0)
```

File: scormxblock/scormxblock.py [hunk 3]
```diff
@@ -411,4 +416,26 @@ def workbench_scenarios():
                 <scormxblock/>
                 </vertical_demo>
              """),
-        ]
\ No newline at end of file
+        ]
+
+    def publish_completion(self):
```
Comment (ziafazal) [inline scormxblock/scormxblock.py]: remove this method.

File: scormxblock/scormxblock.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 import shutil
 import xml.etree.ElementTree as ET
 
+from completion import waffle as completion_waffle
```
Comment (ziafazal) [inline scormxblock/scormxblock.py]: no need to add waffle switches in xblock this should be handled by edx completion tool.

**Review by ziafazal — CHANGES_REQUESTED** [review]
@tehreem-sadat completed first round of review.

**Comment by tehreem-sadat** [comment]
@ziafazal all the changes have been made. Please review it again.

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**